### PR TITLE
Change docs type in insert_many to list of dicts

### PR DESCRIPTION
**Pull request by MezentsevIlya**
Hello! In the `insert_many` method signature docs argument type is `dict`, but actually it must be list of dicts: `list[dict]`

---
**^ Add meaningful description above**
Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/contributing-docs/05_pull_requests.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

File: airflow/providers/mongo/hooks/mongo.py [hunk 1]
```diff
@@ -210,7 +210,7 @@ def insert_one(
         return collection.insert_one(doc, **kwargs)
 
     def insert_many(
-        self, mongo_collection: str, docs: dict, mongo_db: str | None = None, **kwargs
+        self, mongo_collection: str, docs: list[dict], mongo_db: str | None = None, **kwargs
```
Comment (jedcunningham) [inline airflow/providers/mongo/hooks/mongo.py]: Should this be Iterable instead? That's what I see in mongo at least.

Comment (MezentsevIlya) [reply airflow/providers/mongo/hooks/mongo.py]: Thanks, you are right. I changed it to `Iterable`

**Review by potiuk — APPROVED** [review]

**Review by jedcunningham — APPROVED** [review]

**Comment by jedcunningham** [comment]
Static check failure is unrelated (see #37251).

**Comment by jedcunningham** [comment]
Thanks @MezentsevIlya! Congrats on your first commit :tada:

**Pull request merged**